functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,137 @@
+	// Results is used to track labels that have been applied
+	Results fn.Results
+}
+
+// Config parse the functionConfig kubeObject to the fields in the LabelTransformer
+func (p *LabelTransformer) Config(functionConfig *fn.KubeObject) error {
+	// parse labels to NewLabels
+	switch {
+	case functionConfig.IsEmpty():
+		return fmt.Errorf("failed to configure function: `functionConfig` must be either a `ConfigMap` or `SetLabels`")
+	case functionConfig.IsGVK("", "v1", "ConfigMap"):
+		p.NewLabels = functionConfig.NestedStringMapOrDie("data")
+	case functionConfig.IsGVK(fnConfigGroup, fnConfigAPIVersion, fnConfigKind):
+		if _, exist, err := functionConfig.NestedSlice("additionalLabelFields"); exist || err != nil {
+			return fmt.Errorf("`additionalLabelFields` has been deprecated")
+		}
+		p.NewLabels = functionConfig.NestedStringMapOrDie("labels")
+		if len(p.NewLabels) == 0 {
+			return fmt.Errorf("failed to configure function: input label list cannot be empty")
+		}
+	default:
+		return fmt.Errorf("unknown functionConfig Kind=%v ApiVersion=%v, expect `%v` or `ConfigMap`",
+			functionConfig.GetKind(), functionConfig.GetAPIVersion(), fnConfigKind)
+	}
+	// add default fields
+	if err := p.addDefaultLabelFields(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *LabelTransformer) Transform(objects fn.KubeObjects) error {
+	for _, o := range objects {
+		for _, sp := range p.FieldSpecs {
+			if (sp.Group == "" && sp.Version == "" && sp.Kind == "") || o.IsGVK(sp.Group, sp.Version, sp.Kind) {
+				// generate msg
+				res, _ := json.Marshal(p.NewLabels)
+				newResult := fn.Result{
+					Message:     "set labels: " + string(res),
+					Severity:    "",
+					ResourceRef: nil,
+					Field: &fn.Field{
+						Path:          sp.Path,
+						CurrentValue:  nil,
+						ProposedValue: nil,
+					},
+					File: &fn.File{
+						Path:  o.PathAnnotation(),
+						Index: o.IndexAnnotation(),
+					},
+					Tags: nil,
+				}
+				err := updateLabels(o, sp.Path, p.NewLabels, sp.CreateIfNotPresent)
+				if err != nil {
+					return err
+				}
+				p.Results = append(p.Results, &newResult)
+			}
+		}
+	}
+	return nil
+}
+
+func updateLabels(o *fn.KubeObject, fieldPath string, newLabels map[string]string, create bool) error {
+	//TODO: should support user configurable field for labels
+	basePath := strings.Split(fieldPath, "/")
+	keys := make([]string, 0)
+	for k := range newLabels {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for i := 0; i < len(keys); i++ {
+		key := keys[i]
+		val := newLabels[key]
+		newPath := append(basePath, key)
+		_, exist, err := o.NestedString(newPath...)
+		if err != nil {
+			return err
+		}
+		if exist || create {
+			if err = o.SetNestedString(val, newPath...); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+
+}
+
+func (p *LabelTransformer) addDefaultLabelFields() error {
+	var defaultFieldSpecs []FieldSpec
+	err := json.Unmarshal([]byte(commonLabelFieldSpecs), &defaultFieldSpecs)
```

Comment:
Can you add a TODO here to convert commonLabelFieldSpecs to use the IsGVK format?